src/Orleans.TestingHost/TestCluster.cs

```diff
@@ -610,7 +610,13 @@ public async Task<SiloHandle> StartSiloAsync(int instanceNumber, TestClusterOpti
                 InitialData = siloSpecificOptions.ToDictionary()
             });
 
-            var handle = await this.CreateSiloAsync(siloSpecificOptions.SiloName, configurationSources);
+            var configurationBuilder = new ConfigurationBuilder();
+            foreach (var source in configurationSources)
+            {
+                configurationBuilder.Add(source);
+            }
+            var configuration = configurationBuilder.Build();
+            var handle = await this.CreateSiloAsync(siloSpecificOptions.SiloName, configuration);
```

Comment:
I remember that I had to use a list of ConfigurationSource because of serialization issues if I passed IConfiguration directly... didn't you have the same issue? Perhaps now it's fine, just double checking

Reply:
I don't recall the details though... perhaps it's because IConfiguration could be literally anything that fulfills the interface (which might not serialize fine across app domains, or could have unexpected behavior for some users expecting to control that configuration from within the test after it's been sent over the wire). Where as concrete implementations of ConfigurationSource generally serialize fine, and are less prone to misinterpretation once they are sent over the wire.

Reply:
I recall something along those lines. There may have been a later change to the IConfiguration library to add the AsEnumerable() method that we use here to make sure that this is serializable: we end up serializing `KeyValuePair<string, string>[]`, which will not cause an issue.

Users shouldn't expect to be able to control their config after it's been set, especially not in a test cluster. I don't think the old version of this handled that well, anyway. It could only work for file sources, but IIRC they weren't serializable (and they aren't designed to be - we shouldn't be trying to serialize such types)

Reply:
ah, perfect then, certainly the `AsEnumerable()` method didn't exist before, we were asking for something like that in the past, so I guess they finally added it. And I didn't see the usage here as it was merged before I could review :)
Good job, I'm glad using processes is a thing now, and looks like the abstraction pretty much held up :)